### PR TITLE
Update the Plugin documentation

### LINKED ISSUE #6161: Use only `web3` as dependency at chainlink plugin and update the Plugin documentation at `web3.js`
- Update ` web3` versions at the chainlink plugin.
 - Use only` web3` as dependency at the chainlink plugin.
- Update the Plugin documentation accordingly. Which involves:
  1) Only `web3` module is used.
  2) Module Augmentation has been simplified.

### PR DESCRIPTION
**Pull request by Muhammad-Altabba**
## Description

Update the Plugin documentation so only web3 is used and the Module Augmentation is simplified 

Fixes #6161

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

**Review by avkos — APPROVED** [review]

**Review by jdevcs — APPROVED** [review]

File: docs/docs/guides/web3_plugin_guide/plugin_authors.md [hunk 1]
```diff
@@ -214,22 +210,17 @@ public link(parentContext: Web3Context) {
 
 ## Setting Up Module Augmentation
 
-In order to provide type safety and IntelliSense for your plugin when it's registered by the user, you must [augment](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation) the `Web3Context` module. In simpler terms, you will be making TypeScript aware that you are modifying the interface of `Web3Context`, and any class that extends it, to include the interface of your plugin (i.e. your plugin's added methods, properties, etc.). A good tutorial that further explains the topic can be found [here](https://www.digitalocean.com/community/tutorials/typescript-module-augmentation).
+To ensure type safety and enable IntelliSense for your plugin (which still needs to be registered by the user), you must augment the `Web3Context` class inside the `web3` module. In simpler terms, this is to modify the `Web3Context` class, and any inheriting classes, to make your plugin's functionality accessible from within. As a result, your plugin object will be accessible within a namespace of your choice, which will be available within any `Web3Context` object.
```
Comment (Spacesai1or) [inline docs/docs/guides/web3_plugin_guide/plugin_authors.md:213]: I don't think the rewording of this line reads very well. My preference is to revert it back the original

File: docs/docs/guides/web3_plugin_guide/plugin_authors.md [hunk 2]
```diff
@@ -240,148 +231,70 @@ export class CustomRpcMethodsPlugin extends Web3PluginBase {
...
-}
-
-export { Web3Context };
-```
-
-The following represent what your plugin users would see:
-
-![web3 context augmentation](./assets/web3_context_augmentation.png 'web3Context augmentation')
-
-The above screenshot shows IntelliSense thinking `.customRpcMethods.someMethod` is available to call on the instance of `Web3`, even though the plugin user hasn't registered `CustomRpcMethodsPlugin` - running this code would result in an error.
```
Comment (spacesailor24) [inline docs/docs/guides/web3_plugin_guide/plugin_authors.md:256]: I would consider adding this back in as it demonstrates visually the negative result of module augmentation

File: docs/docs/guides/web3_plugin_guide/plugin_authors.md [hunk 3]
```diff
@@ -214,22 +210,17 @@ public link(parentContext: Web3Context) {
...
 
-The `registerPlugin` method exists on the `Web3Context` class, so any class that `extends Web3Context` has the ability to add your plugin's additional functionality to its interface. By augmenting `Web3Context` to include your plugin's interface, you're essentially providing a blanket augmentation that adds your plugin's interface to **all** Web3 modules that extend `Web3Context` (i.e. `web3`, `web3-eth`, `web3-eth-contract`, etc.).
+### Module Augmentation
 
-:::warning
-By augmenting `Web3Context` (and by extension all class interfaces that extend it), your plugin's interface will show up in things like IntelliSense for **all** Web3 modules that extend `Web3Context`, even if your plugin isn't registered - This is something worth making your users aware of, as they'll only be able to use your plugin if they register it with a Web3 class instance using `.registerPlugin`
-
-For context, here is an example of your plugin's interface showing up in IntelliSense even though your plugin hasn't been registered (the code in this example is further explained in the subsequent sections):
```
Comment (spacesailor24) [inline docs/docs/guides/web3_plugin_guide/plugin_authors.md:226]: Along with the picture mentioned below, I would add this warning back in and it's important for the plugin author to be aware of

File: docs/docs/guides/web3_plugin_guide/plugin_users.md [hunk 4]
```diff
@@ -9,42 +9,55 @@ This guide intends to provide the necessary context for registering plugins with
 
 ## Installing the Plugin
 
-Unless otherwise mentioned by the plugin author, installing a plugin should be as simple as `yarn add plugin-name`. This should add the plugin as a dependency within your `package.json` and the plugin should be available to import within your code.
+Unless otherwise mentioned by the plugin author, installing a plugin should be as simple as `yarn add web3-sample-plugin`. This should add the plugin as a dependency within your `package.json` and the plugin should be available to import within your code.
```
Comment (Spacesai1or) [inline docs/docs/guides/web3_plugin_guide/plugin_users.md:12]: I would change this to use `web3-plugin-sample` as ideally all of our plugins will be prefixed with `web3-plugin`

File: docs/docs/guides/web3_plugin_guide/plugin_users.md [hunk 5]
```diff
@@ -9,42 +9,55 @@ This guide intends to provide the necessary context for registering plugins with
...
 
 ```json
+# package.json
 {
-	"name": "your-package-name",
-	"version": "0.0.1",
+	...
 	"dependencies": {
-		"web3-plugin": "0.1.0"
+		"web3-sample-plugin": "0.1.0"
```
Comment (Spacesai1or) [inline docs/docs/guides/web3_plugin_guide/plugin_users.md:19]: `web3-plugin-sample`

File: docs/docs/guides/web3_plugin_guide/plugin_users.md [hunk 6]
```diff
@@ -9,42 +9,55 @@ This guide intends to provide the necessary context for registering plugins with
...
 ```typescript
-import { Web3PluginBase } from 'web3-core';
+// code written by the plugin **developer**
 
-export class SimplePlugin extends Web3PluginBase {
-	public pluginNamespace = 'simplePlugin';
+import { Web3PluginBase } from 'web3';
 
-	public simpleMethod() {
+export class SamplePlugin extends Web3PluginBase {
```
Comment (Spacesai1or) [inline docs/docs/guides/web3_plugin_guide/plugin_users.md:35]: Could rename this to `PluginSample` to match `web3-plugin-sample`, but I don't it's a must

File: docs/docs/guides/web3_plugin_guide/plugin_users.md [hunk 7]
```diff
@@ -9,42 +9,55 @@ This guide intends to provide the necessary context for registering plugins with
...
 ```
 
+Here is an example of how to register the `SamplePlugin` onto an instance of `Web3`:
+
 ```typescript
+// code written by the plugin **user**
+
 import Web3 from 'web3';
-import SimplePlugin from 'web3-plugin';
+import SamplePlugin from 'web3-sample-plugin';
```
Comment (Spacesai1or) [inline docs/docs/guides/web3_plugin_guide/plugin_users.md:57]: `web3-plugin-sample`

File: docs/docs/guides/web3_plugin_guide/plugin_users.md (hunk 4 above)
Comment (Muhammad-Altabba) [reply docs/docs/guides/web3_plugin_guide/plugin_users.md:12]: Done in #6182

File: docs/docs/guides/web3_plugin_guide/plugin_users.md (hunk 6 above)
Comment (Muhammad-Altabba) [reply docs/docs/guides/web3_plugin_guide/plugin_users.md:35]: Done in #6182

File: docs/docs/guides/web3_plugin_guide/plugin_users.md (hunk 5 above)
Comment (Muhammad-Altabba) [reply docs/docs/guides/web3_plugin_guide/plugin_users.md:19]: Done in #6182

File: docs/docs/guides/web3_plugin_guide/plugin_authors.md (hunk 1 above)
Comment (Muhammad-Altabba) [reply docs/docs/guides/web3_plugin_guide/plugin_authors.md:213]: Done in #6182

File: docs/docs/guides/web3_plugin_guide/plugin_users.md (hunk 7 above)
Comment (Muhammad-Altabba) [reply docs/docs/guides/web3_plugin_guide/plugin_users.md:57]: Done in #6182
